.github/dependabot.yml

```
@@ -1,10 +1,28 @@
 version: 2
 updates:
+# rules for the main branch
 - package-ecosystem: maven
   directory: "/"
   schedule:
     interval: daily
   open-pull-requests-limit: 10
+  labels:
+  - "dependencies"
+  ignore:
+  # Ignore major version updates - these should be made manually
+  - dependency-name: "*"
+    update-types: [ "version-update:semver-major" ]
+  # Ignore formatter dependency versions used in codegen modules, since they are fixed for Java 8 support
+  - dependency-name: "com.coveo:fmt-maven-plugin"
+  - dependency-name: "com.google.googlejavaformat:google-java-format"
+
+# rules for the `3.x` branch
+- package-ecosystem: maven
+  directory: "/"
+  schedule:
+    interval: daily
+  open-pull-requests-limit: 10
+  target-branch: "3.x"
   labels:
     - "dependencies"
```

Maybe it'd be helpful to differentiate the label, similar to when we used to have `2.x dependencies`?
```suggestion
    - "3.x dependencies"
```

---

Thanks for the suggestion, accepted.